i18n techs

arikon opened the pull request.

arikon committed:
bem make: Build localization files for bundles

`i18n` elem for `i-bem` block

See bem/bem-tools#172

arikon committed:
bem make: Build localization files for bundles

- add tanker library to parse xml in tanker format
- add i18n tech module
- add i18n.js tech module

See bem/bem-tools#171

arikon committed:
bem make: Build localization files for bundles

- move i18n js core to `i18n` tech for `i-bem__i18n` elem
- rename `i18n` tech files to .js from .json.js
- support `all.js` i18n file for BEM entities
- build single `all.js` i18n file for bundles using `i18n` tech

See bem/bem-tools#172

arikon commented on blocks-common/i-bem/__i18n/lib/tanker.js:
```diff
@@ -0,0 +1,224 @@
+
+function _keyset(params) {
+    // FIXME: get rid of Yandex.Tanker specifics
+    return ((params.project && params.project === 'tanker') ? 'i-' + params.project + '__' : '') + params.keyset;
```
Меня смущает вот эта строка, специфичная для Танкера.
Для чего её добавили и можно ли без неё обойтись?
Если можно, то что для этого нужно сделать?

/cc @narqo @veged

narqo replied:
Мы выкачиваем ключи Танкера к себе в `i-tanker`. При этом нужно уметь «мапить» обращения к динамическим ключам Танкера на свои кейсеты.

arikon replied:
А что такое «ключи Танкера»? Это какой-то специфичный проект в Танкере, который нужно выкачивать всем?
Можно ли сделать так, чтобы обращение к этим ключам происходило не через project='tanker', а нативно, через keyset='i-tanker'?
Очень хочется избавиться от магического хардкода в этом месте.

arikon committed:
bem make: Build localization files for bundles

Add i18n js core

See bem/bem-tools#172

arikon committed:
bem make: Build localization files for bundles

- i18n tech: move extendLangDecl() to LangsMixin, split it to extendDecl() and extendLangDecl()
- i18n.js tech: build *.[lang].js files with client-side js and i18n data

See bem/bem-tools#172

arikon committed:
bem make: Build localization files for bundles

- i18n.js tech: add lang initialization
- i18n.js tech: add more spacing to gererated js files

See bem/bem-tools#172

arikon committed:
bem make: Build localization files for bundles

i18n.js tech: Create symlink from [bundle].js to [bundle].[def-lang].js

See bem/bem-tools#172

arikon committed:
bemhtml tech: Actualize API, add getBuildSuffixes(), remove storeBuildResults()

arikon committed:
Merge branch '0.2' into i18n

arikon committed:
bem make: Build localization files for bundles

i18n.js tech: Make use of `exports.baseTechName` and `exports.techMixin` tech module API

See bem/bem-tools#172

arikon committed:
bem make: Build localization files for bundles

i18n.js tech: Move current lang initialization code generation from serializeI18nData() to separate serializeI18nInit() method

See bem/bem-tools#172

arikon committed:
bem make: Build localization files for bundles

i18n.js tech: Check for language data existence before call to `serializeI18nData()` method

See bem/bem-tools#171

arikon committed:
Refactor js tech to new API,  add js-i tech

arikon committed:
Hint BEMHTML variable using jsdoc in html tech module

arikon committed:
bem make: Build localization files for bundles

- Refactor html tech to recomended API
- Add i18n.html tech
- i18n tech: Add getDefaultLang() and getSourceSuffix() methods to LangsMixin
- i18n.js tech: Move getSuffixForAll() to LangsMixin as getSourceSuffix(); move serializeI18nData() and serializeI18nInit() to separate lib

See bem/bem-tools#171

arikon committed:
bem make: Build localization files for bundles

- i18n.html tech: Make tech generate many files and symplink
- html tech: Factor out getHtml() method

See bem/bem-tools#171

arikon committed:
i18n.js tech: Code cleanup

Commit:
i18n core: Storing current context while processing dynamic keysets

Commit:
Processing XML-nodes in i18n keys

Commit:
Tests for i18n

arikon committed:
Merge branch '0.2' into i18n

arikon committed:
i18n.html tech: Fix language data collisions
